bundles/tampere/content-editor/view/SideContentEditor.js

```diff
@@ -1089,77 +1095,104 @@ Oskari.clazz.define('Oskari.tampere.bundle.content-editor.view.SideContentEditor
-                        });
+                        me.startNewDrawing();
                 });
             } else {
                 areaButton.addClass("disabled");
             }
 
-            var geomEditButton = $("<div />").addClass('selection-area tool').attr('title', me.loc.tools.geometryEdit);
+            var geomEditButton = jQuery("<div />").addClass('selection-area tool').attr('title', me.loc.tools.geometryEdit);
             if (me.layerGeometryType != null) {
 
                 geomEditButton.on('click', function() {
                     me.drawingActive = true;
                     me.drawToolType = "edit";
                     me.clickedGeometryNumber = null;
-                    if (me.layerGeometries.geometry.id.indexOf("Multi") > -1) {
-                        if (me.layerGeometries.geometry.components != undefined) {
-                            for (var i = 0; i < me.layerGeometries.geometry.components.length; i++) {
-                                if (me.layerGeometries.geometry.components[i].atPoint({lon:me.clickCoords.x,lat:me.clickCoords.y}, 2, 2))
-                                {
-                                    me.clickedGeometryNumber = i;
-                                    break;
+                    var geometryType = me.layerGeometries.geometry.getType();
+                    if (geometryType.indexOf("Multi") > -1) {
+                        if (me.layerGeometries.geometry.getCoordinates() != undefined) {
+                            var exists = false;
+                            for (var i = 0; i < me.layerGeometries.geometry.getCoordinates().length; i++) {
+                                if(geometryType === "MultiPoint") {
+                                    var closestPoint = me.layerGeometries.geometry.getClosestPoint([me.clickCoords.x, me.clickCoords.y]);
+                                    var distance = (
+                                        Math.sqrt(
+                                            Math.pow((me.clickCoords.x - closestPoint[0]), 2) +
+                                            Math.pow((me.clickCoords.y - closestPoint[1]), 2)
+                                        ) / me.sandbox.getMap().resolution
+                                    );
+                                    if(distance <= 2.8) {
+                                        me.clickedGeometryNumber = i;
+                                        break;
+                                    }
+                                } else if(geometryType === "MultiPolygon") {
+                                    if(me.layerGeometries.geometry.intersectsCoordinate([me.clickCoords.x, me.clickCoords.y])) {
+                                        me.clickedGeometryNumber = i;
+                                        break;
+                                    }
+                                } else if(geometryType === "MultiLineString") {
+                                    if(exists) {
```

Comment:
Seems that this `exists` flag is only used for breaking the for loop. Why not just call `break;` in L1179 instead of calling `exists = true;`?